### ISSUE TITLE
mpd : Failed to initialize io_uring: io_uring_queue_init() failed: Cannot allocate memory 

### ISSUE THREAD

**Issue by maartenlangeveld** [MichaIng/DietPi#4690]
system: RPi Zero W (armv6) , DietPi v7.5.2 (Bullseye)

**mpd** mentions two exceptions/errors when running 'service mpd status':

```
2021-08-26 01:39:55 root@M-MUSIC-PLAYER:~# service mpd status
● mpd.service - Music Player Daemon (DietPi)
     Loaded: loaded (/lib/systemd/system/mpd.service; disabled; vendor preset: enabled)
     Active: active (running) since Thu 2021-08-26 01:38:04 CEST; 3min 20s ago
       Docs: man:mpd(1)
             man:mpd.conf(5)
   Main PID: 27630 (mpd)
      Tasks: 6 (limit: 991)
        CPU: 7.080s
     CGroup: /system.slice/mpd.service
             └─27630 /usr/bin/mpd --no-daemon /etc/mpd.conf

Aug 26 01:37:59 M-MUSIC-PLAYER systemd[1]: Starting Music Player Daemon (DietPi)...
Aug 26 01:38:04 M-MUSIC-PLAYER mpd[27630]: Aug 26 01:38 : exception: Decoder plugin 'wildmidi' is unavailable: configuration file does not exist: /etc/timidity/timidity.cfg
Aug 26 01:38:04 M-MUSIC-PLAYER mpd[27630]: Failed to initialize io_uring: io_uring_queue_init() failed: Cannot allocate memory
Aug 26 01:38:04 M-MUSIC-PLAYER systemd[1]: Started Music Player Daemon (DietPi).
```

**Comment by jcorporation** [comment]
This could be the problem: https://github.com/MusicPlayerDaemon/MPD/issues/972

**Comment by MichaIng** [comment]
I can replicate the same error on a regular Bullseye VM install, so it's not related RPi, ARMv6 or any such. But the daemon runs fine. Do you face any issues, actually?

The issue is that by default `ulimit -l` shows that the memlock limit is at 64 KiB, while MPD requires (according to the issue you linked) 64 MiB to use "io_uring", a more efficient API for local file I/O or so. So MPD tells you that it cannot use it, which makes it use a fallback. The memlock limit hasn't changed with Bullseye, but making use of "io_uring" has been introduced with MPD 22, which is available since Bullseye: https://github.com/MusicPlayerDaemon/MPD/blob/v0.22.x/NEWS#L149

The upstream systemd unit already has an entry to raise the memlock limit: https://github.com/MusicPlayerDaemon/MPD/blob/v0.22.x/systemd/system/mpd.service.in#L19-L20
But while Debian updated the MPD version, they seem to have missed to update the systemd unit as well, or decided against raising the limit. I'll have a look into bug reports, else create one to have this sorted.

**Comment by MichaIng** [comment]
Second thing is that `wildmidi` is tried to be loaded, expecting a config file, while only `/etc/timidity/timgm6mb.cfg` by `timgm6mb-soundfont` is present.

The default config of Debian has this plugin disabled already:
```
mpd (0.22.4-1) unstable; urgency=medium

  [ Geoffroy Youri Berret ]
  * Enable explicitly yajl in d/rules
  * Disable decoder/wildmidi in default conf
...
```
Our config does not actively enable it, but having a look into Debian's new default config shows that it needs to be actively disabled:
```
decoder {
        plugin        "wildmidi"
        enabled       "no"
        #config_file "/etc/timidity/timidity.cfg"
}
```
Adding this block to `/etc/mpd.conf` prevents MPD's attempt to start the plugin. I'll check if this is the same on previous Debian versions (that it is loaded by default) and have it disabled in case.

**Comment by maartenlangeveld** [comment]
@MichaIng,
Thanks. Have disabled wildmidi plugin in mpd.conf. Will await debian bug report.

**Comment by MichaIng** [comment]
Oh my goodness, it is our own fault, as we overwrite the systemd unit. I didn't see it as we overwrite the packages original file instead of creating an override config. That doesn't make much sense as a package upgrade reverts this.

I'll change it to a little drop-in config, which contains only the adjustments we do (indeed reasonable ones) and similarly the MPD config file won't be overwritten as a whole anymore, but only individual changes done to the Debian default.

**Comment by MichaIng** [comment]
Fixed with: #4726
